### PR TITLE
[schema] AutoConsume should use the schema associated with messages as both writer and reader schema

### PR DESCRIPTION
**Pull request by sijie**
*Motivation*

AutoConsume should use the schema associated with the messages for decoding the schemas.

*Modifications*

- provide a flag enable or disable using the provided schema as the reader schema
- for AUTO_CONSUME schema, disable usnig the provided schema as the reader schema. so it can use the right
  schema version for decoding messages into right generic records
- provide a few util methods for displaying schema data

### CODE REVIEW

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java [hunk 1]
```diff
@@ -732,11 +732,14 @@ protected void preProcessSchemaBeforeSubscribe(PulsarClientImpl pulsarClientImpl
                 SchemaInfo schemaInfo = schemaInfoProvider.getLatestSchema();
                 if (schemaInfo.getType() != SchemaType.AVRO){
```
Comment (congbobo184) [inline pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java]: I think here can add SchemaType.JSON

Comment (sijie) [reply pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java]: @congbobo184 done

**Comment by codelipenghui** [comment]
run Java8 Tests

**Comment by sijie** [comment]
run java8 tests

**Pull request merged**